plugins/out_s3/s3.c

```diff
@@ -1117,8 +1167,19 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         m_upload->upload_errors += 1;
         /* re-add chunk to list */
         if (chunk) {
-            s3_store_file_unlock(chunk);
             chunk->failures += 1;
+            if (chunk->failures > ctx->ins->retry_limit) {
+                s3_retry_warn(ctx, (char *) chunk->fsf->meta_buf, m_upload->input_name,
+                              chunk->create_time, FLB_FALSE);
+                s3_store_file_delete(ctx, chunk);
+                return FLB_ERROR;
+            }
+            else {
+                s3_retry_warn(ctx, (char *) chunk->fsf->meta_buf, m_upload->input_name,
+                              chunk->create_time, FLB_TRUE);
+                s3_store_file_unlock(chunk);
+                return FLB_RETRY;
+            }
```

Comment:
I'm still confused how this will not cause log duplication.

I feel like in the retry case, logs will be appended again to the s3_file

https://github.com/fluent/fluent-bit/blob/760956f50cdc7eef3047df20d2299202a8c68594/plugins/out_s3/s3_store.c#L187-L194

Am I looking at this wrong?